### PR TITLE
lakectl: add --no-progress flag

### LINKED ISSUE #7732: progress bars can be disabled
It would be nice to be able to disable the progress bars you see with `lakectl local clone` and other bulk operations.

Use case: I have some ML training jobs that run in a Kubernetes cluster. That cluster has logging set up so anything a container logs gets sent to an external logging service. Those services typically expect logs to be in a pretty simple format like 1 line per thing logged or in JSON format. However, the progress bars use special characters (e.g. `\r`) to update the cool little graphs and that confuses the heck out of the log parsers. It's still nice to have logs like `downloaded some/file.txt` but the progress bars are problematic. Ideally you'd detect that the `lakectl` stdout isn't a terminal and if that's the case convert the progress bar logs to "regular logs". If that's not possible a flag like `--disable-progress` would be great.

### PR DESCRIPTION
**Pull request by N-o-Z**
Closes #7732

## Change Description

Add a new flag to lakectl to disable progress bar animation.

### Background

Share context and relevant information for the PR: offline discussions, considerations, design decisions etc.

### Bug Fix

Fix check for case insensitive FS when folder has not been created yet
      
### New Feature

Default behavior was maintained, passing --no-progress bar to relevant commands will disable the progress bar animation

### Testing Details

Tested manually

### Breaking Change?

No

### CODE REVIEW

File: cmd/lakectl/cmd/root.go [hunk 1]
```diff
@@ -144,13 +144,15 @@ var (
...
 const (
-	recursiveFlagName   = "recursive"
-	recursiveFlagShort  = "r"
-	presignFlagName     = "pre-sign"
-	parallelismFlagName = "parallelism"
+	recursiveFlagName     = "recursive"
+	recursiveFlagShort    = "r"
+	presignFlagName       = "pre-sign"
+	parallelismFlagName   = "parallelism"
+	noProgressBarFlagName = "no-progress"
```
Comment (arielshaqed) [inline cmd/lakectl/cmd/root.go:151]: Could we keep it positive? 
```suggestion
	progressBarFlagName = "progress"
```

File: pkg/fileutil/fs_case.go [hunk 2]
```diff
@@ -24,6 +24,13 @@ func IsCaseInsensitiveLocation(fs FS, dirPath string, warnFunc func(string)) (bo
 	if err != nil {
 		return false, fmt.Errorf("generate random name: %w", err)
 	}
+	exists, err := fs.Exists(dirPath)
```
Comment (arielshaqed) [inline pkg/fileutil/fs_case.go:27]: Not part of this pr, I think?

**Review by arielshaqed — APPROVED** [review]
Beast! 
Can we default to not showing progress whenever standard output is not a tty? (Or does progress already about that when it cannot show progress....?)

File: cmd/lakectl/cmd/root.go (hunk 1 above)
Comment (N-o-Z) [reply cmd/lakectl/cmd/root.go:151]: If find the `--no-progress` more understandable, especially when the default is false.
Making it positive will require passing explicitly `--progress false` instead of just `--no-progress`

File: pkg/fileutil/fs_case.go (hunk 2 above)
Comment (N-o-Z) [reply pkg/fileutil/fs_case.go:27]: It's in the PR description (😄). It just bothered me so much and the two changes are small enough to allow myself to push it into the same PR.

**Comment by N-o-Z** [comment]
> Beast! Can we default to not showing progress whenever standard output is not a tty? (Or does progress already about that when it cannot show progress....?)

Couldn't find anything about it in the go-pretty code so I just added this check in our code - Thanks!
